nixos/tests/systemd-confinement.nix

```diff
@@ -48,16 +48,22 @@ import ./make-test-python.nix {
       { config.confinement.mode = "chroot-only";
         testScript = ''
           with subtest("chroot-only confinement"):
-              paths = machine.succeed('chroot-exec ls -1 / | paste -sd,').strip()
-              assert_eq(paths, "bin,nix,run")
+              paths = machine.succeed('chroot-exec ls -Am /').strip()
+              assert_eq(paths, "bin, dev, etc, nix, proc, root, run, sys, usr, var")
```

Comment:
This actually should *not* be like this. In `chroot-only` confinement, there should **only** be `/bin` and `/nix` and even `/run` is **not** expected to be there since it should be the *most* restricted environment. Having all that stuff in there means there is certainly something wrong (and already in the first place, since `/run` is in there, which was introduced in via64556974b6674338a227d559975c9b4710f6e751, maybe @andir can explain why).

Reply:
There might be a misunderstanding here -- yes, these directories now listed, but it's definitely not bind mount from the running systems, it's empty directories. There's a new test to make sure the directories are empty and won't bring in new attack surfaces.

Even run is another run directory that has nothing to do with the real /run -- Looking again there are /run/host/os-release and /run/systemd/incoming in there, but it's a small price to pay if that make DynamicUsers and other more commonly used features work...
(Tried adding them to inaccessible paths: systemd/incoming is necessary in setting up the whole private mounts so we service fails to start, but /run/host can be ignored. That appear to be fragments of the systemd "container interface" we probably don't care about: https://systemd.io/CONTAINER_INTERFACE/ - I personally don't see exposing os-release as such a threat, but I agree we might as well hide it. Incoming is just an empty directory once the service is setup so I guess there's not much to hide anyway)

You can spawn a shell with this command to check manually if tests are a bit unwieldy:
```
systemd-run -t -pPrivateMounts=1  -pRuntimeDirectory=systemd-confinement/test -pRootDirectory=/run/systemd-confinement/test -pUMask=066 -pBindReadOnlyPaths=/nix/store -pInaccessiblePaths=-+/run/systemd-confinement/test -pInaccessiblePaths=-+/dev -pInaccessiblePaths=-+/sys -pEnvironment=PATH=$(realpath /run/current-system/sw/bin) -- $(realpath /run/current-system/sw/bin/bash)
```